misc/metrics/src/swarm.rs

```diff
@@ -123,6 +124,13 @@ impl Metrics {
             Box::new(connections_closed.clone()),
         );
 
+        let connection_duration = Family::new_with_constructor(create_connection_duration_histogram as fn()->Histogram);
+        sub_registry.register(
+            "connection_duration",
```

Comment:
```suggestion
            "connection_establishment_duration",
```

Otherwise I would think this is the time the connection was alive for. (Which by the way would also be a great metric to track.)

Reply:
Opened https://github.com/libp2p/rust-libp2p/issues/3135 for the latter.

Reply:
:100: !!
I had a hard time figuring out how to phrase this. Very much like your suggestion.